maven-model-builder/src/main/java/org/apache/maven/model/interpolation/DefaultModelVersionProcessor.java

```diff
@@ -0,0 +1,69 @@
+package org.apache.maven.model.interpolation;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import java.util.Properties;
+
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+import org.apache.maven.model.building.ModelBuildingRequest;
+
+/**
+ * Maven default implementation of the {@link ModelVersionProcessor} to support
+ * <a href="https://maven.apache.org/maven-ci-friendly.html">CI Friendly Versions</a>
+ */
+@Named
+@Singleton
+public class DefaultModelVersionProcessor
+    implements ModelVersionProcessor
+{
+
+    private static final String SHA1_PROPERTY = "sha1";
+
+    private static final String CHANGELIST_PROPERTY = "changelist";
+
+    private static final String REVISION_PROPERTY = "revision";
+
+    @Override
+    public boolean isValidProperty( String property )
+    {
+        return REVISION_PROPERTY.equals( property ) || CHANGELIST_PROPERTY.equals( property )
+            || SHA1_PROPERTY.equals( property );
+    }
+
+    @Override
+    public void overwriteModelProperties( Properties modelProperties, ModelBuildingRequest request )
+    {
+        if ( request.getSystemProperties().containsKey( REVISION_PROPERTY ) )
+        {
+            modelProperties.put( REVISION_PROPERTY, request.getSystemProperties().get( REVISION_PROPERTY ) );
+        }
+        if ( request.getSystemProperties().containsKey( CHANGELIST_PROPERTY ) )
+        {
+            modelProperties.put( CHANGELIST_PROPERTY, request.getSystemProperties().get( CHANGELIST_PROPERTY ) );
+        }
+        if ( request.getSystemProperties().containsKey( SHA1_PROPERTY ) )
+        {
+            modelProperties.put( SHA1_PROPERTY, request.getSystemProperties().get( SHA1_PROPERTY ) );
```

Comment:
@rfscholte I believe that this is a bug in our code since those are are supposed to be user properties and not system properties or we need to support both. @laeubi Not a regression from you, but an observation.

Reply:
As far as I understand the idea ias that system properties given on the command line take precedence over everything else.

Reply:
True, but those are not system properties, those are user properties. A common misconception and badly coined in the Maven core codebase. See: https://github.com/apache/maven/blob/4476026c525a22900c6c5e4f0de6f7362cb201f2/maven-embedder/src/main/java/org/apache/maven/cli/MavenCli.java#L1743-L1753

Reply:
Okay I have changed the PR accordingly.

Reply:
> @rfscholte I believe that this is a bug in our code since those are are supposed to be user properties and not system properties or we need to support both. 

It should be at least user properties. That's probably enough, as I expect that's the way CI servers can provide a value.